domain/info.go

```diff
@@ -0,0 +1,173 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package domain
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/coreos/etcd/clientv3"
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/config"
+	"github.com/pingcap/tidb/ddl"
+	"github.com/pingcap/tidb/mysql"
+	"github.com/pingcap/tidb/util/hack"
+	"github.com/pingcap/tidb/util/printer"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/net/context"
+)
+
+const (
+	// ServerInformation store server information such as IP, port and so on.
+	ServerInformation = "/tidb/server/info"
+)
+
+// infoSyncer stores server info to PD when server start and delete when server down.
+type infoSyncer struct {
+	etcdCli *clientv3.Client
+	info    *ServerInfo
+}
+
+// ServerInfo is server static information.
+// It will not update when server running. So please only put static information in ServerInfo struct.
+type ServerInfo struct {
+	ServerVersionInfo
+	ID         string `json:"ddl_id"`
+	IP         string `json:"ip"`
+	StatusPort uint   `json:"status_port"`
+	Lease      string `json:"lease"`
+}
+
+// ServerVersionInfo is the server version and git_hash.
+type ServerVersionInfo struct {
+	Version string `json:"version"`
+	GitHash string `json:"git_hash"`
+}
+
+// NewInfoSyncer return new infoSyncer. It export for tidb-test test.
+func NewInfoSyncer(id string, etcdCli *clientv3.Client) *infoSyncer {
+	return &infoSyncer{
+		etcdCli: etcdCli,
+		info:    getServerInfo(id),
+	}
+}
+
+func getServerInfo(id string) *ServerInfo {
+	cfg := config.GetGlobalConfig()
+	info := &ServerInfo{
+		ID:         id,
+		IP:         cfg.AdvertiseAddress,
+		StatusPort: cfg.Status.StatusPort,
+		Lease:      cfg.Lease,
+	}
+	info.Version = mysql.ServerVersion
+	info.GitHash = printer.TiDBGitHash
+	return info
+}
+
+//GetServerInfo gets self server static information.
+func (is *infoSyncer) GetServerInfo() *ServerInfo {
+	return is.info
+}
+
+// GetOwnerServerInfoFromPD gets owner server static information from PD.
+func (is *infoSyncer) GetOwnerServerInfoFromPD(ownerID string) (*ServerInfo, error) {
```

Comment:
I think we don't need "ToPD" or "FromPD". The functions in this file are all used to interact with PD.